include/ignition/rendering/Node.hh

```diff
@@ -302,6 +308,29 @@ namespace ignition
       /// \brief Remove all child nodes from this node
       /// This detaches all the child nodes but does not destroy them
       public: virtual void RemoveChildren() = 0;
+
+      /// \brief A map of custom key value data
+      protected: std::map<std::string, Variant> userData;
```

Comment:
This class doesn't have any other protected members yet. I wonder if we should make this private and PIMPLize the class instead. What do you think, @iche033 ?

I'm also not sure if it's best to put this in `BaseNode`, like it was in `BaseVisual`.

Reply:
hmm let's put it in `BaseNode`. That's a templated class so probably won't be able to pimpl'ize it.

Reply:
done https://github.com/ignitionrobotics/ign-rendering/pull/358/commits/92c7ca15a10d05981fc22eb7a8a31a0e95ab95a2